server/src/test/java/org/elasticsearch/index/store/FsDirectoryFactoryTests.java

```diff
@@ -180,4 +184,16 @@ private void doTestStoreDirectory(Path tempDir, String typeSettingValue, IndexMo
             }
         }
     }
+
+    private static Directory unwrapDirectory(Directory directory) {
+        var d = directory;
+        while (d != null) {
+            if (d instanceof FilterDirectory fd) {
+                d = fd.getDelegate();
+            } else {
+                break;
+            }
+        }
+        return d;
+    }
```

Comment:
I think this can be replaced by `FilterDirectory#unwrap`.

Reply:
That's nice!